### PR TITLE
Minor improvements

### PR DESCRIPTION
**Pull request by okurz**
* lockapi: Simplify record_info call
* t: Improve needle test to handle wrong test execution

### CODE REVIEW

File: needle.pm [hunk 1]
```diff
@@ -80,7 +80,7 @@ sub new {
         catch {
             warn "broken json $jsonfile: $_";
         };
-        return undef unless $json;
+        return undef unless $json && $json eq 'HASHREF';
```
Comment (Martchus) [inline needle.pm:83]: Where does `HASHREF` come from? This looks similar to the usual `ref $json eq 'HASH'` check. Did you meant that?

**Review by Martchus — CHANGES_REQUESTED** [review]

File: needle.pm (hunk 1 above)
Comment (okurz) [reply needle.pm:83]: yes, I did. But even if I change it now tests fail and I remember at the time of writing tests could not pass. I will remove that commit.

File: lockapi.pm [hunk 2]
```diff
@@ -73,7 +73,7 @@ sub _log {
         $autotest::current_test->remove_last_result;
         $subject .= ' ' . int($args{amend} / ONE_MINUTE) . 'm' . $args{amend} % ONE_MINUTE . 's';
     }
-    testapi::record_info $subject, $msg;
+    record_info $subject, $msg;
```
Comment (Martchus) [inline lockapi.pm:76]: You will need to import that function if you call it without package name. However, I'm not sure how this is simplifying things.

**Review by Martchus — CHANGES_REQUESTED** [review]

File: lockapi.pm (hunk 2 above)
Comment (okurz) [reply lockapi.pm:76]: Yeah, this is just a ridiculous commit. At the time I am sure it had more meat :)

**Pull request closed without merging**